### PR TITLE
proxy: Remove oxyproxy, CILIUM_USE_ENVOY.

### PR DESCRIPTION
**Pull request by jrajahalme**
Now that we always test with Envoy we can remove the oxy proxy and
simplify by removing the cilium-agent command line argument
"envoy-proxy" and the use of the environment variable
"CILIUM_USE_ENVOY".

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

```release-note
Remove oxyproxy and make Envoy the default proxy
```

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

**Comment by aanm** [comment]
@jrajahalme if the tests fail because of `21-ct-cleanup.sh` feel free to comment out the tests dedicated to envoy.

**Comment by jrajahalme** [comment]
@aanm Have not been successful getting any testing on the CI yet, but locally ```21-ct-cleanup.sh``` succeeds :-)

**Comment by jrajahalme** [comment]
test-me-please

**Review by rlenglet — APPROVED** [review]

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
sigh, Jenkins GitHub Pull Request builder is misbehaving again. I'll try pushing a new PR of this change myself to see if it will trigger, as it has worked fine for me. Sorry about this :(

**Comment by ianvernon** [comment]
Update: pushed to new branch and triggered build on this PR: #2626 . The build links on this (#2625) refer to the same builds as those on #2626.

**Comment by manalibhutiyani** [comment]
This is great! Thanks for the change.

**Review by manalibhutiyani — APPROVED** [review]
Reviewed only the proxy redirect part. Not the ct tests.

**Comment by jrajahalme** [comment]
test-me-please

File: daemon/main.go [hunk 1]
```diff
@@ -336,9 +335,6 @@ func init() {
 	flags.String("enable-policy", endpoint.DefaultEnforcement, "Enable policy enforcement")
 	flags.BoolVar(&enableTracing,
 		"enable-tracing", false, "Enable tracing while determining policy (debugging)")
-	flags.BoolVar(&useEnvoy,
```
Comment (aanm) [inline daemon/main.go]: What happens to current users that are running cilium with `--envoy-proxy`? I think this flag should still exist either with `(deprecated)` and we remove it on the next version or we market it as an hidden flag in viper.

**Review by aanm — CHANGES_REQUESTED** [review]

File: daemon/main.go (hunk 1 above)
Comment (tgraf) [reply daemon/main.go]: I think removal of the flag is safe pre 1.0. Are you worried about the error if the flag is still passed in?

Comment (aanm) [reply daemon/main.go]: correct, there will be an error like:
```
$ cilium-agent --envoy-proxy
Error: unknown flag: --envoy-proxy
```

Comment (tgraf) [reply daemon/main.go]: I'm fine to keep it around for another release but we should remove it pre 1.0.

Comment (jrajahalme) [reply daemon/main.go]: @aanm Do we have an example of how to hide the flag somewhere?

Comment (jrajahalme) [reply daemon/main.go]: I'll try with ```flags.MarkHidden("envoy-proxy")```

Comment (aanm) [reply daemon/main.go]: yes, I think the `MarkHidden` does the trick we are using for `cmdref`

**Review by aanm — APPROVED** [review]

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Bash build failed:
```
01:54:30 ----- 01-guestbook-test.sh: main: running test: 01-guestbook-test
01:54:31 ----- 01-guestbook-test.sh: main/k8s_apply_policy: Current policy revisions:
01:54:31   cilium-plx8j: 11
01:54:31   cilium-tc0tv: 8
01:54:31 networkpolicy "guestbook-redis-deprecated" created
01:54:31 ----- 01-guestbook-test.sh: main/k8s_apply_policy: Waiting for agent cilium-plx8j endpoints to get to revision 12
01:54:31 error: unable to upgrade connection: container not found ("cilium-agent")
01:54:31 ----- 01-guestbook-test.sh: main/k8s_apply_policy/finish_test: finishing test: 01-guestbook-test
```

Not sure why this happened. This happened in the Ginkgo tests as well: https://github.com/cilium/cilium/issues/2422 

Re-trying again.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Due to flakiness with the trigger-phrase, I manually replayed the build on Jenkins: https://jenkins.cilium.io/job/Cilium-Bash-Tests/216/

**Review by raybejjani — APPROVED** [review]
approved for docs

**Comment by tgraf** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]
LGTM! Hooray!!

**Review by tgraf — APPROVED** [review]

**Pull request merged**